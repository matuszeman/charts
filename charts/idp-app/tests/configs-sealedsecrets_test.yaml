suite: configs-sealedsecrets
templates:
  - configs.yaml
  - deployment.yaml
values:
  - values/minimum.yaml
tests:
  - it: when sealedSecret is specified (with content too), then generate SealedSecret, do not generate ConfigMap
    values:
      - values/deployment.yaml
    set:
      containers:
        app:
          configDirs:
            sealed-secret: /sealed-secret
          configFiles:
            sealed-secret:
              key1: /tmp/secret.txt

      configs:
        sealed-secret:
          annotations:
            some: annotation
          sealedSecret:
            scope: namespace-wide
            encryptedData:
              key1: encryptedvalue
          content:
            example.txt: example content
    asserts:
      - matchSnapshot:
          template: configs.yaml
  - it: when sealedSecret with scope is specified, then set it to SealedSecret
    set:
      configs:
        sealed-secret:
          sealedSecret:
            scope: namespace-wide
            encryptedData:
              key1: encryptedvalue
    asserts:
      - isSubset:
          path: metadata.annotations
          content:
            sealedsecrets.bitnami.com/namespace-wide: "true"
        template: configs.yaml
        documentIndex: 0
  - it: when restartPodOnUpdate, then generate Deployment pod with config annotation
    values:
      - values/deployment.yaml
    set:
      configs:
        # below should be added to pod annotations
        sealed-secret:
          restartPodOnUpdate: true
          sealedSecret:
            encryptedData:
              key1: encryptedvalue
    asserts:
      - equal:
          path: spec.template.metadata.annotations
          value:
            config-sealed-secret-hash: 11fd718
        template: deployment.yaml
        documentIndex: 0
  - it: when sealed secret with annotations
    set:
      configs:
        sealed-secret:
          annotations:
            some: annotation
          sealedSecret:
            encryptedData:
              key1: encryptedvalue1
    asserts:
      - equal:
          path: metadata.annotations
          value:
            some: annotation
        template: configs.yaml
        documentIndex: 0
  - it: when sealedSecret with NO scope, then do not set it to SealedSecret
    set:
      configs:
        sealed-secret:
          sealedSecret:
            #scope: namespace-wide
            encryptedData:
              key1: encryptedvalue1
    asserts:
      - notExists:
          path: metadata.annotations
        template: configs.yaml
        documentIndex: 0
      - notExists:
          path: spec.template.metadata.annotations
        template: configs.yaml
        documentIndex: 0