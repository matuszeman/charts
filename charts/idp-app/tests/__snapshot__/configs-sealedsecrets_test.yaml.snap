when sealedSecret is specified (with content too), then generate SealedSecret, do not generate ConfigMap:
  1: |
    apiVersion: bitnami.com/v1alpha1
    kind: SealedSecret
    metadata:
      annotations:
        sealedsecrets.bitnami.com/namespace-wide: "true"
        some: annotation
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: app-name
        app.kubernetes.io/version: 0.1.0
      name: RELEASE-NAME-config-sealed-secret
    spec:
      encryptedData:
        key1: encryptedvalue
      template:
        metadata:
          annotations:
            sealedsecrets.bitnami.com/namespace-wide: "true"
            some: annotation
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: app-name
            app.kubernetes.io/version: 0.1.0
          name: RELEASE-NAME-config-sealed-secret
  2: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: app-name
        app.kubernetes.io/version: 0.1.0
      name: RELEASE-NAME
    spec:
      replicas: 1
      revisionHistoryLimit: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: app-name
      template:
        metadata:
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: app-name
        spec:
          containers:
            - env:
                - name: sealed-secret-env
                  valueFrom:
                    secretKeyRef:
                      key: key1
                      name: RELEASE-NAME-config-sealed-secret
              image: 123456.dkr.ecr.eu-central-1.amazonaws.com/test-server:0.1.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /
                  port: app
              name: app
              ports:
                - containerPort: 80
                  name: app
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /
                  port: app
              resources:
                requests:
                  cpu: 100m
                  memory: 128Mi
              volumeMounts:
                - mountPath: /sealed-secret
                  name: config-sealed-secret
                  readOnly: true
                - mountPath: /tmp/secret.txt
                  name: config-sealed-secret
                  readOnly: true
                  subPath: key1
          enableServiceLinks: false
          nodeSelector: null
          securityContext:
            sysctls:
              - name: net.ipv4.ip_unprivileged_port_start
                value: "0"
          serviceAccountName: RELEASE-NAME
          tolerations: null
          topologySpreadConstraints:
            - labelSelector:
                matchLabels:
                  app.kubernetes.io/instance: RELEASE-NAME
                  app.kubernetes.io/name: app-name
              maxSkew: 1
              topologyKey: topology.kubernetes.io/zone
              whenUnsatisfiable: ScheduleAnyway
            - labelSelector:
                matchLabels:
                  app.kubernetes.io/instance: RELEASE-NAME
                  app.kubernetes.io/name: app-name
              maxSkew: 1
              topologyKey: kubernetes.io/hostname
              whenUnsatisfiable: ScheduleAnyway
          volumes:
            - name: config-sealed-secret
              secret:
                secretName: RELEASE-NAME-config-sealed-secret
